pokecrystalvc.patch.template

```diff
@@ -0,0 +1,734 @@
+﻿/* vim:set noeol nofixeol ff=dos:
+
+This file is preprocessed by tools/make_patch
+It reads the name of a label in the [] of this file,
+  finds its offset in the patched ROM,
+  and replaces any "command" that follows in this file with the appropriate values.
+
+Of note is that the preprocessor is very stupid,
+  and won't check if the offsets in the original ROM differ with those
+  in the patched ROM. It also won't tell you if you didn't catch all of
+  the differences between the ROMs.
+
+The structure of a command is the following:
+  {{Command:argument1:argument2...}}
+
+The possible commands are as follows:
+  Address:
+    Replaced by the address of the label.
+      argument 1: Offset from label. Can be anything parsable with strtol(3).
+      argument 2: Amount of zeroes to prefix it with...
+
+  Patch:
+    Replaced by the byte found at the label in the patched ROM.
+      argument 1: Offset from label. Can be anything parsable with strtol(3).
+      argument 2: If "big", writes a string of bytes starting from the label until it finds a byte that matches with the original ROM.
```

Comment:
Instead of an offset and an optional `big`, this should always take a label name at which to end. My understanding is that the "offset" exists so e.g. `and 1` can be patched to `and 2` by skipping over the identical $E6 opcode byte with `{{patch:1}}`. Instead, just do `{{patch:Foo}}` with:

```
    vc_patch Foo
if DEF(_CRYSTAL_VC)
    and 2
else
    and 1
endc
    vc_patch_end Foo
```

Reply:
Oh, actually, if the `offset` is because the actual patch doesn't cover the $E6 opcode in the first place, then it can stay, but `big` should still take a definite end label.

Reply:
What about having big look for a matching end label, instead of specifying the name? That's what I had in mind.

Maybe we can get rid of offsets by looking for the first differing byte, and the last differing byte within the region.

Reply:
It's theoretically possible for a patch to declare bytes `01 02 03 04`, even if the unpatched ROM contained bytes `01 f2 f3 04`, right? So I'd like `make_patch` to only require the start and end of the region to be patched, without caring about the content.

Reply:
In https://github.com/pret/pokecrystal/pull/882/commits/fb7c391167c09c97cc208255c85fb94d068afd50 `make_patch.c` `big` now takes in the symbols corresponding to the macros `vc_patch` and `vc_patch_end`